lib/src/main/java/graphql/nadel/validation/NadelVirtualTypeValidation.kt

```diff
@@ -0,0 +1,277 @@
+package graphql.nadel.validation
+
+import graphql.nadel.Service
+import graphql.nadel.definition.hydration.isHydrated
+import graphql.nadel.definition.renamed.isRenamed
+import graphql.nadel.definition.virtualType.isVirtualType
+import graphql.nadel.engine.util.unwrapAll
+import graphql.nadel.validation.NadelTypeWrappingValidation.Rule.LHS_MUST_BE_LOOSER_OR_SAME
+import graphql.nadel.validation.NadelTypeWrappingValidation.Rule.LHS_MUST_BE_STRICTER_OR_SAME
+import graphql.nadel.validation.hydration.NadelHydrationValidation
+import graphql.schema.GraphQLArgument
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLObjectType
+
+private class NadelVirtualTypeValidationContext {
+    private val visited: MutableSet<Pair<String, String>> = mutableSetOf()
```

Comment:
Used to keep track of which pair of virtual + backing types we've visited.

To avoid stackoverflows from cyclic references.